src/main/java/picard/sam/ValidateSamFile.java

```diff
@@ -133,75 +133,122 @@ public static void main(final String[] args) {
-        }
-        if (IS_BISULFITE_SEQUENCED) {
-            validator.setBisulfiteSequenced(IS_BISULFITE_SEQUENCED);
-        }
-        if (VALIDATE_INDEX) {
-            validator.setIndexValidationStringency(VALIDATE_INDEX ? IndexValidationStringency.EXHAUSTIVE : IndexValidationStringency.NONE);
-        }
-        if (IOUtil.isRegularPath(INPUT)) {
-            // Do not check termination if reading from a stream
-            validator.validateBamFileTermination(INPUT);
-        }
+            if (IGNORE_WARNINGS) {
+                validator.setIgnoreWarnings(IGNORE_WARNINGS);
+            }
+            if (MODE == Mode.SUMMARY) {
+                validator.setVerbose(false, 0);
+            } else {
+                validator.setVerbose(true, MAX_OUTPUT);
+            }
+            if (IS_BISULFITE_SEQUENCED) {
+                validator.setBisulfiteSequenced(IS_BISULFITE_SEQUENCED);
+            }
+            if (VALIDATE_INDEX) {
+                validator.setIndexValidationStringency(VALIDATE_INDEX ? IndexValidationStringency.EXHAUSTIVE : IndexValidationStringency.NONE);
+            }
+            if (IOUtil.isRegularPath(INPUT)) {
+                // Do not check termination if reading from a stream
+                validator.validateBamFileTermination(INPUT);
+            }
 
-        result = false;
+            result = false;
 
-        switch (MODE) {
-            case SUMMARY:
-                result = validator.validateSamFileSummary(samReader, reference);
-                break;
-            case VERBOSE:
-                result = validator.validateSamFileVerbose(samReader, reference);
-                break;
+            switch (MODE) {
+                case SUMMARY:
+                    result = validator.validateSamFileSummary(samReader, reference);
+                    break;
+                case VERBOSE:
+                    result = validator.validateSamFileVerbose(samReader, reference);
+                    break;
+            }
+            out.flush();
+
+            if (result) {
+                return ReturnTypes.SUCCESSFUL.value();  // ran successfully with no warnings or errors
+            } else {
+                if (validator.getNumErrors() == 0) {
+                    if (validator.getNumWarnings() > 0) {
+                        return ReturnTypes.WARNINGS.value();   // warnings but no errors
+                    }
+                } else {
+                    if (validator.getNumWarnings() > 0) {
+                        return ReturnTypes.ERRORS_WARNINGS.value();   // errors and warnings
+                    } else {
+                        return ReturnTypes.ERRORS.value();  // errors but no warnings
+                    }
+                }
+            }
+        } catch (Exception e) {
+            return ReturnTypes.FAILED.value();  // failed to complete execution
         }
-        out.flush();
 
-        return result ? 0 : 1;
+        return ReturnTypes.SUCCESSFUL.value();
```

Comment:
The only way you get to this line is that `result==false` even though `getNumWarnings()==0 && getNumErrors()==0`.

Since this is a inconsistent state, I would consider that a failure to execute correctly. I would log an error stating this and return FAILED